tools/c7n_azure/tests/test_vm.py

```diff
@@ -53,3 +53,114 @@ def test_find_running(self):
         })
         resources = p.run()
         self.assertEqual(len(resources), 1)
+
+    @arm_template('vm.json')
+    def test_stop(self):
+        p = self.load_policy({
```

Comment:
In this case I think you might consider mocking instead, it would let you do all restart/delete/stop/start easily and I don't think we lose anything - because we just care that the SDK call was made for real with the right parameter.  

With what you have we'd need to make other changes because your impacting that vm provisioned from vm.json so it'll break other tests that use that VM if we were to try to provision and re-record all the tests, so we'd need new VM's just for these, etc.

I'd experiment with `@patch('azure.mgmt.compute.ComputeManagementClient.VirtualMachines')` and then after running your single policy attempt something like `themock.assert_called_with(expected params)` to make sure it tried to call the Start/Stop/Delete method with the right VM name.  Not sure exactly the best place to patch there but its worth experimenting with.

For sure you could add to each little action class a method that does the actual SDK call, like `def stop(...)` and patch that method easily.